idaes_examples/notebooks/docs/flowsheets/temperature_swing_adsorption/temperature_swing_adsorption.ipynb

```diff
@@ -0,0 +1,645 @@
+    "        \"N2\": 0.00960*0.88,\n",
+    "        \"O2\": 0.0},\n",
+    "    \"temperature\": 300.0,\n",
+    "    \"pressure\": 1.0e5}\n",
+    "for i in m.fs.tsa.component_list:\n",
+    "    m.fs.tsa.inlet.flow_mol_comp[:, i].fix(flue_gas[\"flow_mol_comp\"][i])\n",
+    "m.fs.tsa.inlet.temperature.fix(flue_gas[\"temperature\"])\n",
+    "m.fs.tsa.inlet.pressure.fix(flue_gas[\"pressure\"])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 2.3: Fix DOF of TSA unit\n",
+    "\n",
+    "The degrees of freedom of the TSA unit model are: adsorption and desorption temperatures, temperatures of heating and cooling fluids, column diameter, and column  height. These variables must be fixed to solve a square problem."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "The DOF of the TSA unit is 0\n"
+     ]
+    }
+   ],
+   "source": [
+    "# fix design and operating variables of tsa unit - baseline case from Joss et al. 2015\n",
+    "m.fs.tsa.temperature_desorption.fix(430)\n",
+    "m.fs.tsa.temperature_adsorption.fix(310)\n",
+    "m.fs.tsa.temperature_heating.fix(440)\n",
+    "m.fs.tsa.temperature_cooling.fix(300)\n",
+    "m.fs.tsa.bed_diameter.fix(3/100)\n",
+    "m.fs.tsa.bed_height.fix(1.2)\n",
+    "\n",
+    "\n",
+    "# check the degrees of freedom\n",
+    "DOF = degrees_of_freedom(m)\n",
+    "print(f\"The DOF of the TSA unit is {DOF}\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 2.4: Scaling Unit Models\n",
+    "\n",
+    "Creating well scaled models is important for increasing the efficiency and reliability of solvers. Depending on unit models, variables and constraints are often badly scaled. IDAES unit models contain a method to scale variables and constraints to improve solver convergence. To apply the scaled factors defined in each unit model, we need to call the IDAES method `calculate_scaling_factors` in `idaes.core.util.scaling`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# scaling factors\n",
+    "iutil.scaling.calculate_scaling_factors(m.fs.tsa)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 2.5: Define  Solver and Solver Options\n",
+    "\n",
+    "We select the solver that we will be using to initialize and solve the flowsheet."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# define solver and solver options\n",
+    "solver = \"ipopt\"\n",
+    "solver_options = {\n",
+    "    \"mu_init\": 1e-8,\n",
+    "    \"bound_push\": 1e-6,\n",
+    "    \"nlp_scaling_method\": \"user-scaling\",\n",
+    "    \"tol\": 1e-6,\n",
+    "    'halt_on_ampl_error': 'yes',\n",
```

Comment:
1. The `solver="ipopt"` doesn;t actually do much here. Do we actually need it?
2. You should not have `halt_on_ampl_error" in a public facing document - that should be only used for debugging.
3. If possible, you should avoid changing the `mu_init` and `bound_push` settings as these are important for optimization. If the model does not solve without changing these, then it generally implies you have a poorly scaled problem.
4. A reference for the future, but we are moving away from using IPOPTs user scaling as it results in tools and models that work only for IPOPT. We will be deprecating `calculate_scaling_facotrs` in the not too distant future, and switching to use Pyomo's scaling transformation instead.

Reply:
removed 1-3